.travis.yml

```diff
@@ -1,40 +1,51 @@
-language: php
+version: ~> 1.0
 
-dist: xenial
-
-services:
-  - mysql
-  - postgresql
+import:
+  - silverstripe/silverstripe-travis-shared:config/provision/self.yml
 
 env:
   global:
-    - COMPOSER_ROOT_VERSION=4.x-dev
-
-matrix:
+    - COMPOSER_ROOT_VERSION="4.7.x-dev"
+    - REQUIRE_EXTRA="silverstripe-themes/simple:~3.2.0 silverstripe/admin:1.7.x-dev silverstripe/versioned:1.7.x-dev mikey179/vfsstream:^1.6"
+    - PHPUNIT_SUITE="recipe-core"
+
+# Using a manual matrix to exlude PHPUNIT_COVERAGE_TEST which was having an unknown error
+# https://travis-ci.com/github/silverstripe/recipe-core/jobs/457587137
+jobs:
+  fast_finish: true
   include:
     - php: 7.1
-      env: DB=MYSQL COMPOSER_ARG=--prefer-lowest
+      env:
+        - DB=MYSQL
+        - PHPUNIT_TEST=1
     - php: 7.2
-      env: DB=PGSQL
+      env:
+        - DB=PGSQL
+        - PHPUNIT_TEST=1
+        - COW_TEST=1
     - php: 7.3
-      env: DB=MYSQL
+      env:
+        - DB=MYSQL
+        - PDO=1
+        - PHPUNIT_TEST=1
+        - PHPCS_TEST=1
     - php: 7.4
-      env: DB=MYSQL
+      env:
+        - DB=MYSQL
+        #- PHPUNIT_COVERAGE_TEST=1
+        - PHPUNIT_TEST=1
+    - php: nightly
+      env:
+        - DB=MYSQL
+        - PHPUNIT_TEST=1
+        - COMPOSER_INSTALL_ARG=--ignore-platform-reqs
+  allow_failures:
     - php: nightly
-      env: DB=MYSQL COMPOSER_ARG=--ignore-platform-reqs COMPOSER_REQS=marcj/topsort:^2@dev
 
 before_script:
-  - phpenv rehash
-  - phpenv config-rm xdebug.ini
-
-  - composer validate
-  - composer require --no-update silverstripe-themes/simple:~3.2.0 silverstripe/admin:1.x-dev silverstripe/versioned:1.x-dev mikey179/vfsStream:^1.6 $COMPOSER_REQS
-  - if [[ $DB == PGSQL ]]; then composer require silverstripe/postgresql:2.x-dev --no-update; fi
-  - composer update --prefer-source --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile $COMPOSER_ARG
-
-  # Validate cow schema
-  - composer global require silverstripe/cow dev-master $COMPOSER_ARG
-  - ~/.config/composer/vendor/bin/cow schema:validate
-
-script:
-  - vendor/bin/phpunit --testsuite recipe-core
+  # marcj/topsort:^2@dev is a temporary requirement for PHP8 support
+  # This requirement haven't been shipped stable yet
+  # When it is stable then composer.json for silverstripe/config should be updated to ^2
+  # and this line should be removed
+  # ^1 is included because ^2@dev has a minimum php requirement of PHP 7.3
+  - if [ $(php -r 'echo (int) version_compare(phpversion(), "8.0.0", ">=");') = "1" ] ; then compose require marcj/topsort:^2@dev; fi
```

Comment:
This can be dropped now, as `marcj/topsort ^2` has shipped a stable release.

Reply:
Have removed